pymc3/data.py

```diff
@@ -127,109 +128,146 @@ class Minibatch(tt.TensorVariable):
     >>> assert str(x.dtype) == 'int32'
 
-    or simpler with default random seed = 42
+
+    Or, more simply, we can use the default random seed = 42
     `[size, size]`
+
     >>> x = Minibatch(data, batch_size=[10, 10])
 
-    x is a regular :class:`TensorVariable` that supports any math
+
+    In the above, `x` is a regular :class:`TensorVariable` that supports any math operations:
+
+
     >>> assert x.eval().shape == (10, 10)
 
-    You can pass it to your desired model
+    
+    You can pass the Minibatch `x` to your desired model:
+
     >>> with pm.Model() as model:
     ...     mu = pm.Flat('mu')
     ...     sd = pm.HalfNormal('sd')
     ...     lik = pm.Normal('lik', mu, sd, observed=x, total_size=(100, 100))
 
+
     Then you can perform regular Variational Inference out of the box
+    
+
     >>> with model:
     ...     approx = pm.fit()
 
-    Notable thing is that :class:`Minibatch` has `shared`, `minibatch`, attributes
-    you can call later
+
+    Important note: :class:`Minibatch` has `shared`, and `minibatch` attributes
+    you can call later:
+
     >>> x.set_value(np.random.laplace(size=(100, 100)))
 
     and minibatches will be then from new storage
     it directly affects `x.shared`.
-    the same thing would be but less convenient
+
+    A less convenient convenient, but more explicit, way to achieve the same
+    thing:
+
     >>> x.shared.set_value(pm.floatX(np.random.laplace(size=(100, 100))))
 
-    programmatic way to change storage is as follows
-    I import `partial` for simplicity
+
+    The programmatic way to change storage is as follows
+    (I import `partial` for simplicity):
+
+
     >>> from functools import partial
     >>> datagen = partial(np.random.laplace, size=(100, 100))
     >>> x = Minibatch(datagen(), batch_size=10, update_shared_f=datagen)
     >>> x.update_shared()
 
-    To be more concrete about how we get minibatch, here is a demo
-    1) create shared variable
-    >>> shared = theano.shared(data)
+    To be more concrete about how we create a minibatch, here is a demo:
+    1. create a shared variable
+
+        >>> shared = theano.shared(data)
+
+    2. take a random slice of size 10:
+
+        >>> ridx = pm.tt_rng().uniform(size=(10,), low=0, high=data.shape[0]-1e-10).astype('int64')
 
-    2) create random slice of size 10
-    >>> ridx = pm.tt_rng().uniform(size=(10,), low=0, high=data.shape[0]-1e-10).astype('int64')
+    3) take the resulting slice:
 
-    3) take that slice
-    >>> minibatch = shared[ridx]
+        >>> minibatch = shared[ridx]
 
-    That's done. Next you can use this minibatch somewhere else.
-    You can see that implementation does not require fixed shape
-    for shared variable. Feel free to use that if needed.
+    That's done. Now you can use this minibatch somewhere else.
+    You can see that the implementation does not require a fixed shape
+    for the shared variable. Feel free to use that if needed.
```

Comment:
"feel free to use the created minibatch where it fits your needs"

Reply:
Probably worth noting that Minibatch should not go along with most MCMC algos and mostly for VI